### PR TITLE
Add IvdcmProxy and stubs for GPB comunication

### PR DESCRIPTION
**Pull request by bonewell**
- IvdcmProxy
- GpbDataSenderReceiver

### CODE REVIEW

File: src/components/ivdcm_module/include/ivdcm_module/ivdcm.h [hunk 1]
```diff
@@ -81,13 +86,14 @@ class Ivdcm : public functional_modules::GenericModule {
  private:
   static const functional_modules::ModuleID kCANModuleID = 404;
   functional_modules::PluginInfo plugin_info_;
+  IvdcmProxy proxy;
```
Comment (VladSemenyuk) [inline src/components/ivdcm_module/include/ivdcm_module/ivdcm.h]: codestyle

File: src/components/ivdcm_module/include/ivdcm_module/ivdcm_proxy.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#define SRC_COMPONENTS_IVDCM_MODULE_INCLUDE_IVDCM_MODULE_IVDCM_PROXY_H_
+
+#include "ivdcm_module/gpb_data_sender_receiver.h"
+
+namespace ivdcm_module {
+class IvdcmProxyListener;
+
+class IvdcmProxy {
+ public:
+  explicit IvdcmProxy(IvdcmProxyListener *listener);
```
Comment (VladSemenyuk) [inline src/components/ivdcm_module/include/ivdcm_module/ivdcm_proxy.h]: Maybe better to use shared pointer?

**Comment by VladSemenyuk** [comment]
Reviewed

File: src/components/ivdcm_module/include/ivdcm_module/ivdcm.h (hunk 1 above)
Comment (bonewell) [inline src/components/ivdcm_module/include/ivdcm_module/ivdcm.h]: I agree. I will fix in next pull request about GpbDataSenderReceiver.

File: src/components/ivdcm_module/include/ivdcm_module/ivdcm_proxy.h (hunk 2 above)
Comment (bonewell) [inline src/components/ivdcm_module/include/ivdcm_module/ivdcm_proxy.h]: It doesn't necessary. Listener is Ivdcm class which owns IvdcmProxy.

Comment (VladSemenyuk) [inline src/components/ivdcm_module/include/ivdcm_module/ivdcm_proxy.h]: ok

File: src/components/ivdcm_module/src/ivdcm.cc [hunk 3]
```diff
@@ -46,7 +45,9 @@ CREATE_LOGGERPTR_GLOBAL(logger_, "IVDCM")
 
 PLUGIN_FACTORY(Ivdcm)
 
-Ivdcm::Ivdcm() : GenericModule(kCANModuleID) {
+Ivdcm::Ivdcm()
+    : GenericModule(kCANModuleID),
```
Comment (DKlimenko) [inline src/components/ivdcm_module/src/ivdcm.cc]: Worng IDName

File: src/components/ivdcm_module/src/ivdcm.cc [hunk 4]
```diff
@@ -46,7 +45,9 @@ CREATE_LOGGERPTR_GLOBAL(logger_, "IVDCM")
...
+      proxy(IvdcmProxy(this)) {
```
Comment (DKlimenko) [inline src/components/ivdcm_module/src/ivdcm.cc]: bad idea to create proxy with this

File: src/components/ivdcm_module/src/ivdcm_proxy.cc [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+#include "ivdcm_module/ivdcm_proxy_listener.h"
+#include "utils/logger.h"
+
+namespace ivdcm_module {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "IVDCM")
+
+IvdcmProxy::IvdcmProxy(IvdcmProxyListener *listener)
+    : listener_(listener),
+      gpb_(GpbDataSenderReceiver(this)) {
```
Comment (DKlimenko) [inline src/components/ivdcm_module/src/ivdcm_proxy.cc]: please initialise pointer inside constructor

**Comment by DKlimenko** [comment]
reviewed

File: src/components/ivdcm_module/src/ivdcm_proxy.cc (hunk 5 above)
Comment (bonewell) [inline src/components/ivdcm_module/src/ivdcm_proxy.cc]: gpb_ is not pointer.

File: src/components/ivdcm_module/src/ivdcm.cc (hunk 4 above)
Comment (bonewell) [inline src/components/ivdcm_module/src/ivdcm.cc]: Can you explain why?

File: src/components/ivdcm_module/src/ivdcm.cc (hunk 3 above)
Comment (bonewell) [inline src/components/ivdcm_module/src/ivdcm.cc]: :) you are sharp eye

**Pull request merged**